From 69d302a81a2284de17d6e5b6ae8d2c35123f5b97 Mon Sep 17 00:00:00 2001
From: Benjamin Lin <benjamin-jw.lin@mediatek.com>
Date: Fri, 9 May 2025 15:37:25 +0800
Subject: [PATCH 105/107] mtk: mac80211: copy protocol info to non-first
 fragments

Copy protocol info from first fragment to remaining ones.
With protocol info, MT76 may let EAPOL fragments bypass SDO admission control to prevent connection failure.

Signed-off-by: Benjamin Lin <benjamin-jw.lin@mediatek.com>
---
 net/mac80211/tx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index d369677f..470528ac 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -934,6 +934,7 @@ static int ieee80211_fragment(struct ieee80211_tx_data *tx,
 		skb_copy_queue_mapping(tmp, skb);
 		tmp->priority = skb->priority;
 		tmp->dev = skb->dev;
+		tmp->protocol = skb->protocol;
 
 		/* copy header and data */
 		skb_put_data(tmp, skb->data, hdrlen);
-- 
2.45.2

